base/applications/mspaint/winproc.cpp

```diff
@@ -575,6 +577,26 @@ LRESULT CMainWindow::OnSysColorChange(UINT nMsg, WPARAM wParam, LPARAM lParam, B
     return 0;
 }
 
+static CStringW IntEscapeURLW(LPCWSTR psz)
+{
+    // Convert to UTF-8
+    char szUtf8[256];
+    ::WideCharToMultiByte(CP_UTF8, 0, psz, -1, szUtf8, _countof(szUtf8), NULL, NULL);
+    int cchUtf8 = ::lstrlenA(szUtf8);
+
+    // Convert to Base64
+    DWORD cchRet;
+    CStringW strRet;
+    if (!::CryptBinaryToStringW((const BYTE*)szUtf8, cchUtf8, CRYPT_STRING_BASE64, NULL, &cchRet))
+        return strRet;
+    ::CryptBinaryToStringW((const BYTE*)szUtf8, cchUtf8, CRYPT_STRING_BASE64,
```

Comment:
Seriously. This is MSPAINT, not a webbrowser or a mail client here.
The shell function UtfEscape didn't work for those strings, or what?

Reply:
Maybe it's possible to use some APIs of sendmail shell extension?